Sources/PointFree/Auth.swift

```diff
@@ -49,6 +51,20 @@ public func currentUserMiddleware<A>(_ conn: Conn<StatusLineOpen, A>)
     return user.map { conn.map(const($0 .*. conn.data)) }
 }
 
+public func requireLoggedOutUser<A>(
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, A, Data>
+  ) -> Middleware<StatusLineOpen, ResponseEnded, T2<Database.User?, A>, Data> {
```

Comment:
I like how the type signature encodes how we are dropping `Database.User?` from the tuple to force a logged out user